core/nginx/letsencrypt.py

```diff
@@ -47,13 +47,24 @@ def format_for_nginx(fullchain, output):
         for cert in certs[:-1] if len(certs)>2 and os.getenv('LETSENCRYPT_SHORTCHAIN', default="False") else certs:
             pem.write(cert)
 
+def add_DANE_pin(chain, output):
+    with open(output, 'w') as pem:
+        with open(chain, 'r') as chain:
+            for line in chain:
+                pem.write(line)
```

Comment:
No need to do this line by line. 
pem.write(chain.read())
would do the same. 

Is it certain there is a newline at eof? Otherwise appending the second Certificate below could fail.

Reply:
In fact since we're doing SPKI pinning we can use the cross-signed version of the root... so we don't need this at all.